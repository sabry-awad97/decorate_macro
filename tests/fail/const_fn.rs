use decorate_macro::decorate;

fn test_decorator<F, R>(f: F) -> R
where
    F: FnOnce() -> R,
{
    f()
}

#[decorate(test_decorator)]
const fn constant_fn(x: i32) -> i32 {
    x + 1
}

fn main() {}
