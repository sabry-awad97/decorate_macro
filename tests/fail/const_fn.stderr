error: Cannot decorate const functions
  --> tests/fail/const_fn.rs:11:1
   |
11 | const fn constant_fn(x: i32) -> i32 {
   | ^^^^^

error: The decorate attribute cannot be used with const functions
  --> tests/fail/const_fn.rs:11:1
   |
11 | const fn constant_fn(x: i32) -> i32 {
   | ^^^^^
